app/js/ASpellWorker.js

```diff
@@ -130,11 +130,16 @@ class ASpellWorker {
 
     this.pipe.stdout.setEncoding('utf8') // ensure utf8 output is handled correctly
     var output = ''
-    const endMarker = new RegExp('^[a-z][a-z]', 'm')
-    this.pipe.stdout.on('data', chunk => {
-      output = output + chunk
-      // We receive the language code from Aspell as the end of data marker
-      if (chunk.toString().match(endMarker)) {
+    const endMarkerRegex = new RegExp('^[a-z][a-z]', 'm')
```

Comment:
for `lastIndex` to work, you need to specify `g` as well - otherwise it is ignored. See:
```JavaScript
function test(str, regex, flags) {
  r = new RegExp(regex, flags)

  console.log("test with", flags, [
    {result: r.test(str), index: r.lastIndex},
    {result: r.test(str), index: r.lastIndex},
    {result: r.test(str), index: r.lastIndex}
  ])
}

const foo = "foo\nfoo"
const search = '^f'

test(foo, search, 'm')
test(foo, search, 'mg')
```
```
test with m [ { result: true, index: 0 },
  { result: true, index: 0 },
  { result: true, index: 0 } ]
test with mg [ { result: true, index: 1 },
  { result: true, index: 5 },
  { result: false, index: 0 } ]
```
and if you explicitly specify `lastIndex` on an `m` regex, it is ignored.

Reply:
Thanks,  I will fix this.  As soon as I saw the pull reminder coming back from you I knew I should have tested that bit!